src/realm.h

```diff
@@ -362,6 +362,7 @@ typedef enum realm_property_flags {
     RLM_PROPERTY_NULLABLE = 1,
     RLM_PROPERTY_PRIMARY_KEY = 2,
     RLM_PROPERTY_INDEXED = 4,
+    RLM_PROPERTY_FULLTEXT_INDEXED = 8,
```

Comment:
This API makes it possible to construct a schema with both a full text and general index, but that is not something that core supports (currently). Can you add some schema validation and tests for to make sure this is prevented with a nice error message?

Reply:
Well - the interface I have proposed in this PR will not allow to add both types of indexes.